### ISSUE TITLE
Create top miners endpoint

### ISSUE THREAD

**Issue by janmichek** [aeternity/ae_mdw#1963]
- fixed list of:
  { hash: 'ak_wM8yFU8eSETXU7VSN48HMDmevGoCMiuveQZgkPuRn1nTiRqyv', name: 'WoolyPooly' },
  { hash: 'ak_dArxCkAsk1mZB1L9CX3cdz1GDN4hN84L3Q8dMLHN4v8cU85TF', name: '2miners' },
  { hash: 'ak_9MsbDuBTtKegKpj5uSxfPwmJ4YiN6bBdtXici682DgPk8ycpM', name: '2miners.solo' },

- fixed time window of 24h
- blocks mined count per entry
- reference https://www.aeknow.org/miner
- data will be used for table and chart in aescan as in reference

_Originally posted by @janmichek in https://github.com/aeternity/ae_mdw/issues/1943#issuecomment-2399159977_

**Comment by janmichek** [comment]
@yaboiishere I am afraid the result is something else that I requested.


This seems like grouped by time (day), which is not the same as 24hour window. Right?

**Comment by yaboiishere** [comment]
Won't thss work as well? Im cautious of implementing something like that for the actual last 24 hours since that would require going over all data points and summing it up. We would prefer not doing that since if there are too many records it will result into having slow endpoints that might even timeout (an example of this is the bug that appeared in the testnet for the stats endpoint, where because of that reason it timed out).

Also we do actually know how many blocks can happen in a day for the mainnet/testnet, but implementing this would be a problem for hyperchains since there you can configure way more blocks than that, which would make the endpoint slow/unresponsive

**Comment by janmichek** [comment]
Yeah, so what can I do... I am developer, not a manager. I have been waiting for over 3 months for rejecting message. Nevermind I will code it on frontend

**Comment by yaboiishere** [comment]
I've given it more thought over the weekend and I've realised that hyperchains don't have miners they have validators, so the problem I had with implementing this the way you want now should be possible without slowing down the hc. This endpoint would only be available to parent chains though (i.e. PoW ones)

**Comment by janmichek** [comment]
Ok, good. So when could you deliver it?

**Comment by janmichek** [comment]
@yaboiishere ?

**Comment by yaboiishere** [comment]
Hey, sorry I've missed this. I'm currently working on another task, but in the meantime I've starting planning on how the change woul look like. If the change goes as planned and there are no major issues with the mdw on hyperchains I hope to have it developed in the next few days
